### PR TITLE
Princeton handler

### PR DESCRIPTION
**Pull request by tacaswell**
attn @stuwilkins

### CODE REVIEW

**Comment by danielballan** [comment]
Needs a test.

**Comment by tacaswell** [comment]
The downside of commenting on a per-commit basis is that GH won't do old-comment folding for us.

Think I addressed everything, fixed a few bugs and got the npy writer tests running again.

**Comment by tacaswell** [comment]
```
00:09 $ python run_tests.py
...............................
Name                      Stmts   Miss  Cover   Missing
-------------------------------------------------------
filestore.api                 3      0   100%   
filestore.commands           51      0   100%   
filestore.conf               16      0   100%   
filestore.file_writers       56      3    95%   129-133
filestore.handlers           97     14    86%   21-24, 29-43
filestore.odm_templates      28      0   100%   
filestore.readers             0      0   100%   
filestore.readers.spe       162    126    22%   49-52, 56-100, 106, 109, 113, 117, 121-122, 126, 130, 134-139, 143, 147, 151, 159-162, 165-166, 169-170, 173, 176, 180-203, 206-213, 216-218, 221-235, 238-240, 245-247
filestore.retrieve           53      0   100%   
filestore.utils               0      0   100%   
-------------------------------------------------------
TOTAL                       466    143    69%   
----------------------------------------------------------------------
Ran 31 tests in 2.806s

OK
```

Coverage for reference (need to get coveralls turned on on this repo).

**Comment by ericdill** [comment]
Coveralls is running again. It broke when we namechanged from fileStore -> filestore

**Comment by ericdill** [comment]
https://coveralls.io/builds/2041419

**Comment by tacaswell** [comment]
@stuwilkins Take a look at the changes to the SPE handler

File: filestore/handlers.py [hunk 1]
```diff
@@ -1,16 +1,62 @@
...
+            spe_obj = PrincetonSPEFile(fname)
+            self._f_cache[point_number] = spe_obj
+
+        spe = self._f_cache[point_number]
+        data = spe.getData()
+        # if we fix up old entries in CSX FS db we can do va
+        # if data.shape[0] != self._fpp:
+        #     raise IntegrityError(
+        #               "expected {} frames, found {} frames".format(
+        #                      self._fpp, data.shape[0]))
```
Comment (stuwilkins) [inline filestore/handlers.py]: We can make this change. There is no SPE files which matter in the database at this point

**Pull request merged**